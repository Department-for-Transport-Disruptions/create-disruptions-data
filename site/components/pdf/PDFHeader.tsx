import { View, Text, StyleSheet, Font } from "@react-pdf/renderer";

Font.registerHyphenationCallback((word) => [word]);

const PDFStyles = StyleSheet.create({
    container: {
        flexDirection: "row",
        alignItems: "flex-start",
        textAlign: "left",
        fontStyle: "bold",
        borderTopWidth: 0,
        borderRightWidth: 0,
        borderLeftWidth: 0,
        borderBottomWidth: 4,
        margin: 5,
        fontFamily: "Helvetica-Bold",
    },
    column: {
        width: "6.5%",
    },
    title: {
        width: "30.5%",
    },
    date: {
        width: "4.5%",
    },
    status: {
        width: "8.5%",
    },
});

const PDFHeader = () => (
    <View style={PDFStyles.container} fixed>
        <Text style={PDFStyles.column}>id</Text>
        <Text style={PDFStyles.title}>title</Text>
        <Text style={PDFStyles.column}>mode</Text>
        <Text style={PDFStyles.column}>operator wide</Text>
        <Text style={PDFStyles.column}>network wide</Text>
        <Text style={PDFStyles.column}>services affected</Text>
        <Text style={PDFStyles.column}>stops affected</Text>
        <Text style={PDFStyles.date}>start</Text>
        <Text style={PDFStyles.date}>end</Text>
        <Text style={PDFStyles.date}>publish start</Text>
        <Text style={PDFStyles.date}>publish end</Text>
        <Text style={PDFStyles.column}>severity</Text>
        <Text style={PDFStyles.column}>live</Text>
        <Text style={PDFStyles.status}>status</Text>
    </View>
);

export default PDFHeader;
