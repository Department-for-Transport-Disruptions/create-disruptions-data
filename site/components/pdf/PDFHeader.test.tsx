import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import PDFHeader from './PDFHeader';

describe('PDFHeader', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<PDFHeader />);
        expect(asFragment()).toMatchSnapshot();
    });
});
