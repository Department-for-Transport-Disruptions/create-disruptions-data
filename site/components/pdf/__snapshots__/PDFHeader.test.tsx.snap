// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PDFHeader > should render correctly 1`] = `
<VIEW
  fixed={true}
  style={
    {
      "alignItems": "flex-start",
      "borderBottomWidth": 4,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "fontFamily": "Helvetica-Bold",
      "fontStyle": "bold",
      "margin": 5,
      "textAlign": "left",
    }
  }
>
  <TEXT
    style={
      {
        "width": "6.5%",
      }
    }
  >
    id
  </TEXT>
  <TEXT
    style={
      {
        "width": "30.5%",
      }
    }
  >
    title
  </TEXT>
  <TEXT
    style={
      {
        "width": "6.5%",
      }
    }
  >
    mode
  </TEXT>
  <TEXT
    style={
      {
        "width": "6.5%",
      }
    }
  >
    operator wide
  </TEXT>
  <TEXT
    style={
      {
        "width": "6.5%",
      }
    }
  >
    network wide
  </TEXT>
  <TEXT
    style={
      {
        "width": "6.5%",
      }
    }
  >
    services affected
  </TEXT>
  <TEXT
    style={
      {
        "width": "6.5%",
      }
    }
  >
    stops affected
  </TEXT>
  <TEXT
    style={
      {
        "width": "4.5%",
      }
    }
  >
    start
  </TEXT>
  <TEXT
    style={
      {
        "width": "4.5%",
      }
    }
  >
    end
  </TEXT>
  <TEXT
    style={
      {
        "width": "6.5%",
      }
    }
  >
    severity
  </TEXT>
  <TEXT
    style={
      {
        "width": "6.5%",
      }
    }
  >
    live
  </TEXT>
  <TEXT
    style={
      {
        "width": "8.5%",
      }
    }
  >
    status
  </TEXT>
</VIEW>
`;
